### DISCUSSION TITLE
10 instance warning despite disconnecting clients

### DISCUSSION THREAD

**Discussion by aetheryx** [prisma/prisma#11033] [Q&A] [answered]
Example code:
```ts
import { PrismaClient } from '@prisma/client';

(async () => {
  for (let index = 0; index < 11; index++) {
    console.log(index);
    const client = new PrismaClient();
    await client.$connect();
    await client.$disconnect();
  }
})();
```

I'm disconnecting each client, but I'm still getting the warning logged:
> warn(prisma-client) There are already 10 instances of Prisma Client actively running.  

How can I properly clean up a Prisma instance?

**Comment by janpio** [comment, marked as answer]
Disconnecting does not destroy the Client. You should not start them in the first place but reuse the same Prisma Client.

**Reply by aetheryx** [reply]
Is it possible to destroy the client? I'm playing around with hot-reload for a backend application using Prisma. I understand that the client should be re-used in production, hot-reload would only be a development thing

**Reply by janpio** [reply]
I don't know.

But the warning is a warning as it is technically totally ok to do this - it should just not happen in a production environment.

**Reply by aetheryx** [reply]
Gotcha, appreciate the information 👍

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
